internal/tls/utils.go

```diff
@@ -0,0 +1,59 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+)
+
+func ParseCiphers(ciphers []string) ([]uint16, error) {
+	suites := []uint16{}
+
+	cipherMap := map[string]uint16{
+		"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305":    tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+		"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305":  tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+		"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256":   tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+		"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256": tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+		"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384":   tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384": tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+		"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256":   tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+		"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA":      tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+		"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256": tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+		"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA":    tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+		"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA":      tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+		"TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA":    tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+		"TLS_RSA_WITH_AES_128_GCM_SHA256":         tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+		"TLS_RSA_WITH_AES_256_GCM_SHA384":         tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+		"TLS_RSA_WITH_AES_128_CBC_SHA256":         tls.TLS_RSA_WITH_AES_128_CBC_SHA256,
+		"TLS_RSA_WITH_AES_128_CBC_SHA":            tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+		"TLS_RSA_WITH_AES_256_CBC_SHA":            tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+		"TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA":     tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,
+		"TLS_RSA_WITH_3DES_EDE_CBC_SHA":           tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
+		"TLS_RSA_WITH_RC4_128_SHA":                tls.TLS_RSA_WITH_RC4_128_SHA,
+		"TLS_ECDHE_RSA_WITH_RC4_128_SHA":          tls.TLS_ECDHE_RSA_WITH_RC4_128_SHA,
+		"TLS_ECDHE_ECDSA_WITH_RC4_128_SHA":        tls.TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,
+	}
+	for _, cipher := range ciphers {
+		if v, ok := cipherMap[cipher]; ok {
+			suites = append(suites, v)
+		} else {
+			return suites, fmt.Errorf("unsupported cipher %q", cipher)
+		}
+	}
+
+	return suites, nil
+}
+
+func ParseTLSVersion(version string) (uint16, error) {
+	versionMap := map[string]uint16{
+		"SSL30": tls.VersionSSL30,
+		"TLS10": tls.VersionTLS10,
+		"TLS11": tls.VersionTLS11,
+		"TLS12": tls.VersionTLS12,
+		// for go >= 1.12
+		//"TLS13": tls.VersionTLS13,
```

Comment:
This change will actually make it impossible to enable TLS1.3 anymore, not that I'm aware of anyone using it.  However, I think we should split this file into several files `common_go110.go`, `common_go120.go` so we have multiple version of the values and it is still able to set TLS1.3.

Let's do this for not only the TLS version but also the ciphers.

Reply:
I'm not sure that I've changed it with right way. I've create files _go112 and _notgo112 (for build on all other version without tls1.3). Basically as I understand we can create separate source for each version but in that way there will be error while move to 1.13...

Reply:
Sorry, I told you wrong.  We want a `common.go` which will always be compiled, then we also should have a common_go112.go, these version tags work a bit different from the platform tags.  The file gets built when using any version of Go later than 1.12 (so it will get built in 1.13+ without a rename).

So in common.go we should have:
```
cipherMap = map[string]uint16{
	"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305": tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
}
```

Then over in common_go112.go we add in the new values:
```
func init() {
    cipherMap["blah"] = tls.BLAH
}
```

Reply:
Oh, I see. But may be it would be better to have common_go110 and common_go111 that will remove unsupported versions/ciphers? I think it would be simpler to support that in future. How do you think?

Reply:
Sounds good in theory but I'm not sure it's possible, the primary file needs to work in all versions.  Try it out and see how it goes.

Reply:
So my idea isn't working you right because in that case I  should use undeclared constants from crypto/tls... I wanted to have this map as constnt. It it why I've covered it with getter.  
I also double checked, basically with current implementation it should be compiled with any versions, because in notgo112 there is `+build !go1.12` (< go.1.12) and in go112 - `+build go1.12` (>- go.1.12). But if you really want to have common.go it will work as well. It can just confuse that we add something in runtime to constant list...

Reply:
Should I commit changes with this common.go as you offered?:) Or current implementation will work for you?

Reply:
@danielnelson I've pushed changes that was recommended. Could you please review?:)